src/main/java/io/supertokens/storage/mysql/queries/PasswordlessQueries.java

```diff
@@ -181,8 +181,8 @@ public static void createCode(Start start, PasswordlessCode code)
             } catch (SQLException e) {
 
                 String message = e.getMessage();
-                if (message.contains(
-                        "Cannot add or update a child row: a foreign key constraint fails (`supertokens`.`passwordless_codes`, CONSTRAINT `passwordless_codes_ibfk_1` FOREIGN KEY (`device_id_hash`) REFERENCES `passwordless_devices` (`device_id_hash`) ON DELETE CASCADE ON UPDATE CASCA)")) {
+                if (message.contains("foreign key constraint fails") && message.contains("passwordless_codes")
+                        && message.contains("device_id_hash")) {
```

Comment:
- The message check should happen in Start.java and not here
- Please see how we check for foreign key violations in other places. We don't actually check for table names. And if we must do that here, we should check the full table name using `Config.getConfig(start).getPasswordlessCodesTable()`

Reply:
noted.